node/src/command.rs

```diff
@@ -338,6 +340,74 @@ pub fn run() -> Result<()> {
                 ))
             })
         }
+        Some(Subcommand::SoloChain(cmd)) => {
+            // Cannot use create_configuration function because that needs a chain spec.
+            // So write our own `create_runner` function that doesn't need chain spec.
+            let container_chain_cli = cmd.run.normalize();
+            let runner = solochain::create_runner(&container_chain_cli)?;
+
+            // The expected usage is
+            // `tanssi-node solochain --flag`
+            // So `cmd` stores the flags from after `solochain`, and `cli` has the flags from between
+            // `tanssi-node` and `solo-chain`. We are ignoring the flags from `cli` intentionally.
+            // Would be nice to error if the user passes any flag there, but it's not easy to detect.
+
+            // Zombienet appends a --chain flag after "solo-chain" subcommand, which is ignored, so it's fine,
```

Comment:
Why are we ignoring `--chain` command? Can we instead use that to decide which relay chain id and based on that chain spec?

Reply:
This is the `--chain` from container chain args, which should be empty because it will change depending on which container chain the collator is assigned to.

I believe the relay chain id for which I have a TODO is only needed when using an external relay (rpc or light node).

Reply:
So, zombienet appends `--chain` in container chain args correct? 

Okay. In that case why don't we have solochain's `--chain_id` or something (depending upon how relay chain is going to be used.)?

Reply:
But in zombienet we don't need the relay chain id, and outside of zombienet we won't have `--chain`.

Reply:
I am confused. Why not? Can you explain the reason behind both of them?

Reply:
Sure. So in zombienet the relay chain id is obtained from the relay chain args appended by zombienet, so we have these args: `--chain simple-template.json -- --chain rococo-local.json`. And the relay chain id is obtained from the `rococo-local` arg. This will also apply to non-zombienet collators that pass the chain spec of starlight in the future.

Outside of zombienet we don't have the first `--chain`, so the args will be like `-- --chain starlight-spec.json`. And again, we already have the relay chain id from the relay chain spec.

The only case this may break is when using an external rpc as relay chain, so we don't have the starlight chain spec as an argument, but I'm not even sure if that's broken and I don't think it's worth to test it at this stage. If it's broken we can fix it in the future.

Reply:
Make sense. So, how `-- --chain starlight-spec.json` is this handled? 
Since, We have hardcoded solo chain id right now.

Will you be addressing that as part of your todo?

Reply:
Oh my bad, that hardcoded relay chain id is actually not used, you can set it to `""` and it will still work. I can improve the TODO to mention this.

Reply:
But it seems to be used by `RelayChainCli`?

Reply:
It is clear now. Please mention this in the todo as this is not obvious. :)

Reply:
*fixed*